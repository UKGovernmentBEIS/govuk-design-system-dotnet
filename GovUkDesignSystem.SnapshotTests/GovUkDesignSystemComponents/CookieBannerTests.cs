using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class CookieBannerTests : SnapshotTestBase
    {
        private CookieBannerViewModel DefaultCookieBannerViewModel()
        {
            return new CookieBannerViewModel
            {
                BannerState = BannerState.ShowBanner,
                CookieType = CookieType.Analytics,
                UseAntiForgeryToken = false,
                ButtonClickAction = "link_after_accept_or_reject",
                ViewCookiesLink = "link_to_view_cookies",
                ReturnUrl = "original_url"
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("CookieBanner", DefaultCookieBannerViewModel());
        }
        
        [Fact]
        public async Task Render_AdditionalCookies()
        {
            // Arrange
            var viewModel = DefaultCookieBannerViewModel();
            viewModel.CookieType = CookieType.Additional;
            
            // Act & Assert
            await VerifyPartial("CookieBanner", viewModel);
        }
        
        [Fact]
        public async Task Render_AcceptedBanner()
        {
            // Arrange
            var viewModel = DefaultCookieBannerViewModel();
            viewModel.BannerState = BannerState.ShowAccepted;
            
            // Act & Assert
            await VerifyPartial("CookieBanner", viewModel);
        }
        
        [Fact]
        public async Task Render_RejectedBanner()
        {
            // Arrange
            var viewModel = DefaultCookieBannerViewModel();
            viewModel.BannerState = BannerState.ShowRejected;
            
            // Act & Assert
            await VerifyPartial("CookieBanner", viewModel);
        }
        
        [Fact]
        public async Task Render_HiddenBanner()
        {
            // Arrange
            var viewModel = DefaultCookieBannerViewModel();
            viewModel.BannerState = BannerState.Hide;
            
            // Act & Assert
            await VerifyPartial("CookieBanner", viewModel);
        }
    }
}