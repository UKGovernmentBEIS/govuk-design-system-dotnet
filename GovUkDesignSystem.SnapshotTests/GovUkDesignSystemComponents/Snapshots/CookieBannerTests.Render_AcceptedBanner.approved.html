

    <form method="POST" action="link_after_accept_or_reject">
        <input id="ReturnUrl" name="ReturnUrl" type="hidden" value="original_url" />
        <div class="govuk-cookie-banner " data-nosnippet role="region" aria-label="Cookies on our service">
            <div class="govuk-cookie-banner__message govuk-width-container">
                    <div class="govuk-grid-row">
                        <div class="govuk-grid-column-two-thirds">

                            <div class="govuk-cookie-banner__content">
                                <p class="govuk-body">You’ve accepted additional cookies. You can <a class="govuk-link" href="link_to_view_cookies">change your cookie settings</a> at any time.</p>
                            </div>
                        </div>
                    </div>


        <button class="govuk-button "
                data-module="govuk-button"
                value="hide"
                type="submit"
                name="consent"
                
                aria-disabled="false"
                data-prevent-double-click="false"
                >

Hide this message        </button>
            </div>
        </div>

    </form>
